scripts/pre_commit_linter.py

```diff
@@ -1165,76 +1165,6 @@ def __init__(self, all_filepaths, verbose_mode_enabled=False):
-        returns the parsed contents as a Python dictionary.
-        """
-
-        # Select JS files which need to be checked.
-        files_to_check = [
-            filepath for filepath in self.all_filepaths if (
-                filepath.endswith(('.js', '.ts')))
-            and not any(fnmatch.fnmatch(filepath, pattern) for pattern in
-                        EXCLUDED_PATHS)]
-        parsed_js_and_ts_files = dict()
-        if not files_to_check:
-            return parsed_js_and_ts_files
-        compiled_js_dir = tempfile.mkdtemp(
-            dir=os.getcwd(), prefix='tmpcompiledjs')
-        if not self.verbose_mode_enabled:
-            print 'Validating and parsing JS and TS files ...'
-        for filepath in files_to_check:
-            if self.verbose_mode_enabled:
-                print 'Validating and parsing %s file ...' % filepath
-            file_content = FileCache.read(filepath).decode('utf-8')
-
-            try:
-                # Use esprima to parse a JS or TS file.
-                parsed_js_and_ts_files[filepath] = esprima.parseScript(
-                    file_content, comment=True)
-            except Exception as e:
-                # Compile typescript file which has syntax not valid for JS
-                # file.
-                if filepath.endswith('.js'):
-                    shutil.rmtree(compiled_js_dir)
-                    raise Exception(e)
-                try:
-                    compiled_js_filepath = self._compile_ts_file(
-                        filepath, compiled_js_dir)
-                    file_content = FileCache.read(compiled_js_filepath).decode(
-                        'utf-8')
-                    parsed_js_and_ts_files[filepath] = esprima.parseScript(
-                        file_content)
-                except Exception as e:
-                    shutil.rmtree(compiled_js_dir)
-                    raise Exception(e)
-
-        shutil.rmtree(compiled_js_dir)
-
-        return parsed_js_and_ts_files
-
-    def _compile_ts_file(self, filepath, dir_path):
-        """Compiles a typescript file and returns the path for compiled
-        js file.
-        """
-        allow_js = 'true'
-        lib = 'es2017,dom'
-        no_implicit_use_strict = 'true'
-        skip_lib_check = 'true'
-        target = 'es5'
-        type_roots = './node_modules/@types'
-        cmd = (
-            './node_modules/typescript/bin/tsc -outDir %s -allowJS %s '
-            '-lib %s -noImplicitUseStrict %s -skipLibCheck '
-            '%s -target %s -typeRoots %s %s typings/*') % (
-                dir_path, allow_js, lib, no_implicit_use_strict,
-                skip_lib_check, target, type_roots, filepath)
-        subprocess.call(cmd, shell=True, stdout=subprocess.PIPE)
-        compiled_js_filepath = os.path.join(
-            dir_path, os.path.basename(filepath).replace('.ts', '.js'))
-        return compiled_js_filepath
 
     def _lint_all_files(self):
```

Comment:
This method should be broken up and implemented in the child classes, so JS/TS lint in one class, and the everything else in the other class.

Reply:
Done